EIPS/eip-6596.md

```diff
@@ -0,0 +1,717 @@
+          "description": "Flags whether or not the HAT token holder is the copyright owner of the underlying asset. Does not include the copyright to the underlying asset"
+        },
+        "copyrightRightDocumentURI": {
+          "type": "string",
+          "description": "Link to legal document outlining the rights of the token owner. Specific dimensions include the right to display a work via digital and physical mediums, present the work publicly, create or sell copies of the work, and create or sell derivations from the underlying asset.",
+          "format": "uri"
+        },
+        "isOwnerAssetCopyrightRightHolder": {
+          "type": "boolean",
+          "description": "Flags whether or not the HAT token holder is the copyright owner of the underlying asset. Does not include the copyright to the underlying asset"
+        },
+        "assetCopyrightRightDocumentURI": {
+          "type": "string",
+          "description": "Link to legal document outlining the rights of the token owner. Specific dimensions include the right to display a work via digital and physical mediums, present the work publicly, create or sell copies of the work, and create or sell derivations from the HAT.",
+          "format": "uri"
+        },
+        "isOwnerTokenReprintRightHolder": {
+          "type": "boolean",
+          "description": "Flags whether or not the HAT token owner has non-exclusive reprint/second serial rights to the HAT. Does not include the rights to the underlying asset"
+        },
+        "tokenReprintRightDocumentURI": {
+          "type": "string",
+          "description": "Link to legal document outlining the reprint/second serial rights of the token owner"
+        },
+        "isOwnerAssetReprintRightHolder": {
+          "type": "boolean",
+          "description": "Flags whether or not the HAT token owner has non-exclusive reprint/second serial rights to the underlying asset. Does not include the rights to the underlying asset"
+        },
+        "assetReprintRightDocumentURI": {
+          "type": "string",
+          "description": "Link to legal document outlining the reprint/second serial rights of the token owner"
+        },
+        "isEligibleForRelatedTokenAirdrops": {
+          "type": "boolean",
+          "description": "Flags whether or not the HAT token holder is eligible for related HAT airdrops"
+        },
+        "isEligibleForMetaverseAccess": {
+          "type": "boolean",
+          "description": "Flags whether or not the HAT token holder can gain access to the HAT metaverse"
+        }
+      },
+      "required": [
+        "assetType",
+        "id",
+        "issuers",
+        "summary",
+        "issueDate",
+        "editionNumber",
+        "editionCount",
+        "fileURI",
+        "fileSize",
+        "fileFormat",
+        "latestPossibleCreationDate"
+      ]
+    }
+  },
+  "required": [
+    "name",
+    "description",
+    "image",
+    "properties"
+  ]
+}
+```
```

Comment:
(Nit) It's far easier to read this in typescript. Mind porting it?

Reply:
Thanks for the feedback, @Pandapip1.

That's a great idea, actually. I'd like to follow what other metadata extension standard (ERC-4955, ERC-5625) have done and provide a JSON schema to fully describe the metadata extension. This would allow others who want to work with this standard to have a JSON schema for validation.

There appears to have value in both, especially with the JSON Schema providing very structured description of the properties without having to resort to comments to describe the fields. @Pandapip1 happy to follow follow your lead on this, though. What do you think if we provide both TypeScript interface and JSON schema?

Reply:
I personally have used typescrupt in my Metadata extension (I forget the numbef). However, you are free to use whichever well-defined format you choose.

Reply:
Done!